lib/streamlit/elements/data_editor.py

```diff
@@ -593,6 +597,14 @@ def experimental_data_editor(
                 columns_config[INDEX_IDENTIFIER] = {}
             columns_config[INDEX_IDENTIFIER]["hidden"] = True
 
+        # If disabled not a boolean, we assume it is a list of columns to disable.
+        # This gets translated into the columns configuration:
+        if not isinstance(disabled, bool):
```

Comment:
Should we just be checking if it's an `Iterable`? Why do we check if it's not a boolean?

Reply:
Its not super straight forward to check if something is an iterable since there are many iterable types and they don't necessarily have a shared parent type. Best way would be to check it there is an __iter__ attribute, but that is also a bit hacky.

Reply:
Ah okay good to know.